internal/mongo/connect.go

```diff
@@ -0,0 +1,43 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/JakeHumphries/gymshark-hacker-news/internal/models"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+// ConnectDb tests the connection to mongo and returns a mongo client
+func ConnectDb(ctx context.Context, cfg models.Config) (*mongo.Client, error) {
+
+	url := fmt.Sprintf("mongodb://%s:%s@%s:%s", cfg.DatabaseUser, cfg.DatabasePassword, cfg.DatabaseName, cfg.DatabasePort)
```

Comment:
I'd escape the password by doing `url.QueryEscape(cfg.DatabasePassword)` as we've noticed issues with special characters in other projects.